### ISSUE TITLE
[Bug] Cancel Modal created in component in parent

### ISSUE THREAD

**Issue by joehatescoffee** [Blazored/Modal#150]
I recently upgraded BlazoredModal from 3.0 to 4.1.  

I have three razor pages: Search, ImageStack, ImagePopup.

Search has a top search bar that is above all layers. The body reports results of the search.

Search utilizes component ImageStack in the results. 

ImageStack helps me manage some html regarding the images returned in a search.
It is also the component that opens the BlazoredModal with `Modal.Show<ImagePopup>(null, parameters, options);`

In version 3, I could close the modal from search with `Modal.Cancel();` if i had `@inject IModalService Modal;` up top.

I am guessing with the implementation of instances `Modal.Cancel()` on the IModalService no longer exits. However, I cannot find a method to capture all modal instances to close/cancel them all.

To summarize, I want to be able to close all modals when I perform another search or return home. As it is, the modal will continue to stay up with all results appearing in the layers under it.

**Comment by larsk2009** [comment]
Hi @joehatescoffee 

Please have a look at [this](https://modal.blazored.com/usage/long-running-task.html) and/or [this](https://github.com/Blazored/Modal/pull/143)

If that doesn't help, let me know!

**Comment by joehatescoffee** [comment]
I had previously look at the example.  All of them have the close or cancel being called from a component of the page that opened the modal.

My example has the modal being called called from a component of the page and being closed from the Main page.

After some time this morning, I did find a workaround.

I added the line `public static ModalReference modal { get; set; }` to the main page and in the component set the reference using `Main.modal = Modal.Show<ImagePopup>(null, parameters, options);`

Now, I can test the main page for a modal.

**Comment by larsk2009** [comment]
Nice to hear you have found something that works. 

Be careful about one thing though. As far as I am aware, static variables are shared across **ALL** users in Blazor **Server side.** See [this](https://github.com/dotnet/aspnetcore/issues/15451) issue. This means that if you are using server-side Blazor your solution may work for 1 person but not when multiple people are accessing your site because every user will reference to this same variable.

I'll admit I have not tested this yet, but it makes sense to me so it is definitely something you should check.

**Comment by joehatescoffee** [comment]
Thanks for sharing that. I discovered it doesn't really work. (This is only my second project in Blazor)

Using two separate browser sessions, I opened a modal in each. The subsequent open broke the earlier and I could not close it.

I will have to dig deeper into this next week. Other projects are due.

**Comment by chrissainty** [comment]
This sounds like a design problem and not so much an issue with the model. I don't think there is any work for us to do here so I'm going to close this issue.

If you need help with your design please ask here and we will help as much as we can.
